### PR TITLE
#2430, #2431: Merges duplicate style guides; fixes indentation

### LINKED ISSUE #2430: Duplicate Dafny style guides
There is a Dafny style guide in the main Dafny project. There is a modification of that style guide in the libraries project.

Duplicating information leads to maintenance headaches.

I propose consolidating to one style guide (in the main project) that is referenced from the libraries project.

### PR DESCRIPTION
**Pull request by davidcok**
Fixes #2430, #2431

Merges duplicate style guides; fixes indentation

<small>By submitting this pull request, I confirm that my contribution is made under the terms of the [MIT license](https://github.com/dafny-lang/dafny/blob/master/LICENSE.txt).</small>

### CODE REVIEW

File: docs/StyleGuide/Style-Guide.md [hunk 1]
```diff
@@ -18,10 +18,17 @@ Any **lemmas**, **predicates**, **functions**, **methods**, **classes**, **modul
...
 method FindIndex(arr: seq<int>, k: int)
-    ...
+  ...
 ```
 
-Any static or global **constants** are named with `UPPERCASE_WITH_UNDERSCORES`.
+The **lemma** keyword indicates a ghost method used for proof purposes. Any **lemma** names should be prefixed with `Lemma`.
+
+  lemma LemmaValueIsInIndex(arr: seq<int>, k: int)
+    ...
```
Comment (cpitclaudel) [inline docs/StyleGuide/Style-Guide.md:27]:
```suggestion
    lemma LemmaValueIsInIndex(arr: seq<int>, k: int)
      ...
```

File: docs/StyleGuide/Style-Guide.md [hunk 2]
```diff
@@ -18,10 +18,17 @@ Any **lemmas**, **predicates**, **functions**, **methods**, **classes**, **modul
 are named with `PascalCase`.
 ```dafny
 method FindIndex(arr: seq<int>, k: int)
-    ...
+  ...
 ```
 
-Any static or global **constants** are named with `UPPERCASE_WITH_UNDERSCORES`.
+The **lemma** keyword indicates a ghost method used for proof purposes. Any **lemma** names should be prefixed with `Lemma`.
```
Comment (cpitclaudel) [inline docs/StyleGuide/Style-Guide.md:24]: FWIW, I don't follow this convention.  We don't prefix method names with `Method` and function names with `Function`, and `lemma LemmaXYZ` looks very redundant.

File: docs/StyleGuide/Style-Guide.md [hunk 3]
```diff
@@ -31,47 +38,53 @@ Avoid redundant names when variables or methods are in a class/module.
...
-    //
-    // this method name can be simplified to ToString()
-    // so that the method call is Integer.ToString(i)
-    // instead of Integer.IntegerToString(i).
+  // The following method converts the given integer
+  // to a string.
+  //
+  // this method name can be simplified to ToString()
+  // so that the method call is Integer.ToString(i)
+  // instead of Integer.IntegerToString(i).
```
Comment (cpitclaudel) [inline docs/StyleGuide/Style-Guide.md:46]: Should this comment be moved to the prose above?

File: docs/StyleGuide/Style-Guide.md (hunk 2 above)
Comment (davidcok) [reply docs/StyleGuide/Style-Guide.md:24]: I would agree; I was not editing the documents, just merging them.

Comment (davidcok) [reply docs/StyleGuide/Style-Guide.md:24]: I removed it, to fit with the philosophy of not including redundant information in names.

File: docs/StyleGuide/Style-Guide.md (hunk 3 above)
Comment (davidcok) [reply docs/StyleGuide/Style-Guide.md:46]: Done.

File: docs/StyleGuide/Style-Guide.md [hunk 4]
```diff
@@ -293,15 +365,25 @@ Avoid too little or too much whitespace that reduces the overall readability.
...
 ```
 
+#### Attributes
+Omit white space between the `:` and the attribute name.
+```dafny
+// YES
+method {:extern} m() { ... }
+
+// NO
+method {: extern } m() { ... }
```
Comment (cpitclaudel) [inline docs/StyleGuide/Style-Guide.md:384]:
```suggestion
method {: extern} m() { ... }
```

Comment (MikaelMayer) [reply docs/StyleGuide/Style-Guide.md]: Can the space before "extern" also be removed? It's not a type ascription.

File: docs/StyleGuide/Style-Guide.md [hunk 5]
```diff
@@ -85,7 +87,7 @@ import Coffee
 ...
 ```
 
-However, if some members of a module are used very frequently, import it using `opened`:
+However, if some members of a module are used very frequently, import them using `opened`:
```
[the remark this answers is not in the document]
Comment (davidcok) [reply docs/StyleGuide/Style-Guide.md:90]: Ah you are correct. I changed 'it'/'them,' to 'the module'

File: docs/StyleGuide/Style-Guide.md (hunk 4 above)
Comment (davidcok) [reply docs/StyleGuide/Style-Guide.md]: The point is that the space before the extern is not recommended style, which is why the '// NO' on the line before.

**Review by cpitclaudel — APPROVED** [review]

**Pull request merged**